EMS/core-bundle/src/Form/DataField/DataFieldType.php

```diff
@@ -207,15 +199,15 @@ public function isDisabled(array $options): bool
      */
     public function getElasticsearchQuery(DataField $dataField, array $options = []): array
     {
-        throw new \Exception('virtual method should be implemented by child class : '.\get_class($this));
+        return [];
```

Comment:
The 3 parent functions that were throwing an exception: 

- public function **getElasticsearchQuery**(DataField $dataField, array $options = []): array
> only overwritten by DataLinkFieldType and used inside the CriteriaController

- public function **getChoiceList**(FieldType $fieldType, array $choices): array
> only overwritten by DataLinkFieldType and used inside the CriteriaController

- public static function **filterSubField**(array $data, array $option): array
> only overwritten by DateRangeFieldType and used RawDataTransformer: in case of virtualfield wich is not a container (datarange)

Reply:
Making them abstract, will create a lot of overhead, because all or 39 fieldTypes then need to implement it. 
Now the DataFieldType gives a 'default' implementation.